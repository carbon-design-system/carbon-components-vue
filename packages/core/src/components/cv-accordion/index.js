import CvAccordion from './cv-accordion';
import CvAccordionItem from './cv-accordion-item';
import CvAccordionSkeleton from './cv-accordion-skeleton';

export { CvAccordion, CvAccordionItem, CvAccordionSkeleton };
