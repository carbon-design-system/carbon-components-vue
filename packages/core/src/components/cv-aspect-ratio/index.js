import CvAspectRatio from './cv-aspect-ratio';
export { CvAspectRatio };
