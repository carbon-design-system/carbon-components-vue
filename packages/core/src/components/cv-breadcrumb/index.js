import CvBreadcrumb from './cv-breadcrumb';
import CvBreadcrumbItem from './cv-breadcrumb-item';
import CvBreadcrumbSkeleton from './cv-breadcrumb-skeleton';

export { CvBreadcrumb, CvBreadcrumbItem, CvBreadcrumbSkeleton };
