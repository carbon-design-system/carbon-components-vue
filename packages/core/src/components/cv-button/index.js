import CvButton from './cv-button';
import CvIconButton from './cv-icon-button';
import CvButtonSkeleton from './cv-button-skeleton';
import CvButtonSet from './cv-button-set';

export { CvButton, CvIconButton, CvButtonSkeleton, CvButtonSet };
