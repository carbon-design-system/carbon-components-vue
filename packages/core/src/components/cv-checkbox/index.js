import CvCheckbox from './cv-checkbox';
import CvCheckboxSkeleton from './cv-checkbox-skeleton';
export { CvCheckbox, CvCheckboxSkeleton };
