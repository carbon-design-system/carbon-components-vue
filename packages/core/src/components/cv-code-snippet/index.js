import CvCodeSnippet from './cv-code-snippet';
import CvCodeSnippetSkeleton from './cv-code-snippet-skeleton';
export { CvCodeSnippet, CvCodeSnippetSkeleton };
