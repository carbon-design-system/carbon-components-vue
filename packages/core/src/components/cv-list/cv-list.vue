<template>
  <component
    :is="tagType"
    class="cv-list"
    :class="{
      [`${carbonPrefix}--list--nested`]: nested,
      [`${carbonPrefix}--list--ordered`]: internalOrdered,
      [`${carbonPrefix}--list--unordered`]: !internalOrdered,
    }"
    :data-nested="nested"
  >
    <slot></slot>
  </component>
</template>

<script>
import { carbonPrefixMixin } from '../../mixins';
export default {
  name: 'CvList',
  mixins: [carbonPrefixMixin],
  props: {
    ordered: { type: Boolean, default: undefined },
    nested: Boolean,
  },
  computed: {
    internalOrdered() {
      if (this.nested && this.ordered === undefined) {
        if (this.$parent && this.$parent.$parent && this.$parent.$parent.$_CvList) {
          return this.$parent.$parent.internalOrdered;
        }
      }
      return this.ordered;
    },
    tagType() {
      return this.internalOrdered ? 'ol' : 'ul';
    },
  },
  created() {
    this.$_CvList = true;
  },
};
</script>
