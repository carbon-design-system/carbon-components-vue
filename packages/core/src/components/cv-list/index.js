import CvList from './cv-list';
import CvListItem from './cv-list-item';

export { CvList, CvListItem };
