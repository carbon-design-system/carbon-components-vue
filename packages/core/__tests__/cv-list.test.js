// import { shallowMount as shallow, mount } from '@vue/test-utils';
import { testComponent, awaitNextTick } from './_helpers';
const { shallowMount: shallow, mount } = awaitNextTick;
import { CvList, CvListItem } from '@/components/cv-list';

describe('CvList', () => {
  const CvListItemWrapper = {
    render() {
      return (
        <CvList nested={true}>
          <CvListItem>test item</CvListItem>
        </CvList>
      );
    },
  };
  // ***************
  // PROP CHECKS
  // ***************
  testComponent.propsAreType(CvList, ['ordered', 'nested'], Boolean);
  testComponent.propsHaveDefaultOfUndefined(CvList, ['ordered']);

  // ***************
  // SNAPSHOT TESTS
  // ***************

  it('should render correctly when nested and parent is ordered', async () => {
    const propsData = { nested: false, ordered: true };
    const wrapper = await mount(CvList, {
      slots: {
        default: CvListItemWrapper,
      },
      propsData,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render correctly when nested and parent is unordered', async () => {
    const propsData = { nested: false, ordered: false };
    const wrapper = await mount(CvList, {
      slots: {
        default: CvListItemWrapper,
      },
      propsData,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render correctly when ordered', async () => {
    const propsData = { ordered: true, nested: false };
    const wrapper = await shallow(CvList, {
      propsData,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('should render correctly when unordered', async () => {
    const propsData = { ordered: false, nested: false };
    const wrapper = await shallow(CvList, {
      propsData,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render correctly with slots', async () => {
    const wrapper = await shallow(CvList, {
      slots: {
        default: CvListItem,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  // ***************
  // FUNCTIONAL TESTS
  // ***************
  it('`internalOrdered` should be computed correctly when ordered', async () => {
    const propsData = { ordered: true, nested: true };
    const wrapper = await shallow(CvList, {
      propsData,
    });
    expect(wrapper.vm.internalOrdered).toEqual(true);
  });
  it('`internalOrdered` should be computed correctly when unordered', async () => {
    const propsData = { ordered: false, nested: true };
    const wrapper = await shallow(CvList, {
      propsData,
    });
    expect(wrapper.vm.internalOrdered).toEqual(false);
  });

  it('`internalOrdered` should be computed correctly when nested and parent is ordered', async () => {
    const propsData = { nested: false, ordered: true };
    const wrapper = await mount(CvList, {
      slots: {
        default: CvListItemWrapper,
      },
      propsData,
    });
    expect(wrapper.findAll(CvList).at(1).vm.internalOrdered).toEqual(true);
    expect(wrapper.findAll(CvList).at(0).vm.internalOrdered).toEqual(true);
  });

  it('`internalOrdered` should be computed correctly when nested and parent is unordered', async () => {
    const propsData = { nested: false, ordered: false };
    const wrapper = await mount(CvList, {
      slots: {
        default: CvListItemWrapper,
      },
      propsData,
    });
    expect(wrapper.findAll(CvList).at(1).vm.internalOrdered).toEqual(false);
    expect(wrapper.findAll(CvList).at(0).vm.internalOrdered).toEqual(false);
  });

  it('`tagType` should be computed correctly when ordered', async () => {
    const propsData = { ordered: true };
    const wrapper = await shallow(CvList, {
      propsData,
    });
    expect(wrapper.vm.tagType).toEqual('ol');
  });
  it('`tagType` should be computed correctly when unordered', async () => {
    const propsData = { ordered: false };
    const wrapper = await shallow(CvList, {
      propsData,
    });
    expect(wrapper.vm.tagType).toEqual('ul');
  });
});

describe('CvListItem', () => {
  // ***************
  // PROP CHECKS
  // ***************

  // ***************
  // SNAPSHOT TESTS
  // ***************
  it('should render correctly without slots', async () => {
    const wrapper = await shallow(CvListItem);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render correctly with slots', async () => {
    const wrapper = await shallow(CvListItem, {
      slots: {
        default: '<div class="slot class">slot content</div>',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  // ***************
  // FUNCTIONAL TESTS
  // ***************
});
