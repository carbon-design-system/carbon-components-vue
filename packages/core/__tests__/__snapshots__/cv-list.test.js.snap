// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CvList should render correctly when nested and parent is ordered 1`] = `
<ol class="cv-list bx--list--ordered">
  <ol data-nested="true" class="cv-list bx--list--nested bx--list--ordered">
    <li class="cv-list-item bx--list__item">test item</li>
  </ol>
</ol>
`;

exports[`CvList should render correctly when nested and parent is unordered 1`] = `
<ul class="cv-list bx--list--unordered">
  <ul data-nested="true" class="cv-list bx--list--nested bx--list--unordered">
    <li class="cv-list-item bx--list__item">test item</li>
  </ul>
</ul>
`;

exports[`CvList should render correctly when ordered 1`] = `<ol class="cv-list bx--list--ordered"></ol>`;

exports[`CvList should render correctly when unordered 1`] = `<ul class="cv-list bx--list--unordered"></ul>`;

exports[`CvList should render correctly with slots 1`] = `
<ul class="cv-list bx--list--unordered">
  <li class="cv-list-item bx--list__item"></li>
</ul>
`;

exports[`CvListItem should render correctly with slots 1`] = `
<li class="cv-list-item bx--list__item">
  <div class="slot class">slot content</div>
</li>
`;

exports[`CvListItem should render correctly without slots 1`] = `<li class="cv-list-item bx--list__item"></li>`;
